luigi/parameter.py

```diff
@@ -311,13 +311,19 @@ class DateParameter(_DateParameterBase):
     """
 
     date_format = '%Y-%m-%d'
-    _timedelta = timedelta(days=1)
 
-    def parse(self, s):
-        """
-        Parses a date string formatted as ``YYYY-MM-DD``.
-        """
-        return datetime.datetime.strptime(s, self.date_format).date()
+    def next_in_enumeration(self, value):
+        return value + datetime.timedelta(days=self.interval)
+
+    def normalize(self, value):
+        if value is None:
+            return None
+
+        if isinstance(value, datetime.datetime):
+            value = value.date()
```

Comment:
ah yes. finally!

This means that we cna pass datetime's and not only date's as constructors to {Date,Month,Year}Parameter and it would automatically do the right thing. Am I correct?